doc/articles/features/using-skia-hosting-native-controls.md

```diff
@@ -2,38 +2,50 @@
 uid: Uno.Skia.Embedding.Native
 ---
 
-# Embedding Native Controls in Skia Apps
+# Embedding Native Elements in Skia Apps
 
-In a Skia Desktop (Windows) app, you can embed native controls in your Skia app. This is useful if you want to use a native control for a specific task, for instance, to integrate an existing WPF control.
+> [!NOTE]
+> As of Uno Platform 6.0, running on macOS desktop does not yet support native controls embedding.
 
-In principle, the native control hosting places the native control in an overlay on top of the Skia control which renders the whole app's UI. The native control is then rendered by the native windowing system and is not part of the Uno Platform visual tree. That control cannot be styled by Uno Platform styles, and Uno Platform controls cannot be placed on top of native controls.
+In an Uno Platform app with a Skia renderer, i.e. using `net9.0-desktop` or adding `SkiaRenderer` to the `UnoFeatures` MSBuild property, you can embed native controls in your Skia app. This is useful if you want to use a native control for a specific task, for instance, to integrate an existing WPF control.
 
-> [!NOTE]
-> As of Uno Platform 5.2, X11 and macOS targets does not yet support native controls embedding.
+Each target platform has its own idea of a native element.
+
+| Platform                        | Native element                                 | Description                                  |
+|---------------------------------|------------------------------------------------|----------------------------------------------|
+| Skia Desktop (Win32)            | Uno.UI.NativeElementHosting.Win32NativeWindow  | A native Windows window with a unique `Hwnd` |
+| Skia Desktop (WPF)              | System.Windows.UIElement                       | A WPF control                                |
+| Skia Desktop (X11)              | Uno.UI.NativeElementHosting.X11NativeWindow    | A native X11 window with a unique `XID`      |
```

Comment:
macOS hosting is in src\Uno.UI.Runtime.Skia.MacOS\MacOSNativeElementHostingExtension.cs

Reply:
`MacOSNativeElement` is not public and I'm not sure if it works outside of MPE/webview

Reply:
On macOS native elements must be created by native code, either inside our bundled library or _eventually_ a 3rd party one. This is how we can avoid ObjC interop (and using `netX.X-macos`) inside managed code.

By _eventually_ I mean more work is required in order to support 3rd part native controls :)